src/hmem_cuda.c

```diff
@@ -430,38 +454,60 @@ static int cuda_hmem_dl_init(void)
 static void cuda_hmem_dl_cleanup(void)
 {
 #if ENABLE_CUDA_DLOPEN
+	dlclose(cuda_attr.nvml_handle);
 	dlclose(cuda_attr.driver_handle);
 	dlclose(cuda_attr.runtime_handle);
 #endif
 }
 
 static int cuda_hmem_verify_devices(void)
 {
-	cudaError_t cuda_ret;
+	nvmlReturn_t nvml_ret;
+	unsigned int nvml_device_count = 0;
+	
+	/* Check w/ nvmlDeviceGetCount_v2() first, to avoid more expensive
+	 * call to cudaGetDeviceCount() when possible.
+	 */
 
-	/* Verify CUDA compute-capable devices are present on the host. */
-	cuda_ret = ofi_cudaGetDeviceCount(&cuda_attr.device_count);
-	switch (cuda_ret) {
-	case cudaSuccess:
-		break;
+	/* Verify NVIDIA devices are present on the host. */
+	nvml_ret = ofi_nvmlDeviceGetCount_v2(&nvml_device_count);
+	if (NVML_SUCCESS == nvml_ret) {
+		FI_INFO(&core_prov, FI_LOG_CORE,
+			"Number of NVIDIA devices detected: %u\n",
+			nvml_device_count);
 
-	case cudaErrorNoDevice:
-		return -FI_ENOSYS;
+		if (nvml_device_count > 0) {
+			cudaError_t cuda_ret;
```

Comment:
Declare variables at the top of the function.  It makes them easier to find.

Reply:
OK, will update